pkg/agent/cniserver/server.go

```diff
@@ -528,6 +532,17 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (*
 	cniVersion := cniConfig.CNIVersion
 	cniResult, _ := result.Result.GetAsVersion(cniVersion)
 
+	if s.secondaryNetworkEnabled {
+		status, err := netdefutils.CreateNetworkStatus(cniResult, cniConfig.Name, true, nil)
+		if err != nil {
+			klog.ErrorS(err, "Create NetworkStatus failed", "Pod", klog.KRef(podNamespace, podName))
+		} else {
+			if err := k8s.UpdatePodNetworkStatusAnnotation(s.kubeClient, ctx, []netdefv1.NetworkStatus{*status}, podName, podNamespace, true); err != nil {
```

Comment:
I wonder if it would be better to move this logic entirely to the SecondaryNetwork controller. It looks out of place here.
We already send a `PodUpdate` event to the SecondaryNetwork controller for processing. Should we just add the `NetworkStatus` (for the primary interface) to the `PodUpdate` struct (it can be `nil` when `secondaryNetworkEnabled` is false), and let that controller handle it? I do not like the way `UpdatePodNetworkStatusAnnotation` is implemented at the moment, and in particular the `isPrimary` parameter.

cc @tnqn @luolanzone